### PR TITLE
Adding example code for exporting vm metadata and re-importing it wit…

### PR DESCRIPTION
**Pull request by thomasmck**
…h new VDIs

Signed-off-by: Thomas Mckelvey <thomas.mckelvey@citrix.com>

### CODE REVIEW

**Comment by chandrikas** [comment]
LGTM

**Comment by olivierlambert** [comment]
Hey @chandrikas quick question while you are here ;)

Will CBT support delta VHD export? That would be far easier for us to just remove VDI data and keep metadata, then ask the diff between this VDI metadata and new VDI.

**Comment by chandrikas** [comment]
Hi @olivierlambert, CBT is image format agnostic at the moment and just works with raw blocks. No promises at the moment but it should be possible in the future.

**Comment by olivierlambert** [comment]
Damn, that would have been 1000 times easier for us to keep the delta VHD export :(

Are you agree to say it won't solve the "half of space" usage issue: on a non-thin provisioned storage, if you want to backup ALL guests at once, you need to use less than half of the SR space, otherwise you won't be able to snapshot them all.

The "only" (but good!) new thing is to allow discarding snapshot data **after** the snapshot creation.

Would it be possible to create directly a "metadata" only snapshot without using data space?

In the end, our current delta backup solution on thin pro SR answers all the need. The only restriction is when SR is on a non thin based SR :/

**Comment by thomasmck** [comment]
As per discussion with @robertbreker I will split this out into 2 files and integrate into the end to end flow so that it is more representative.

File: cbt_vm_metadata_import_export.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+"""
+For a given vm and old/new VDI pairs this script will export the metadata of
+a VM then reimport it with new VDIs. This script should be run when you want
+to make a new VM from a backup. The new VDIs containing the backup data should
+be created before this script is run.
```
[the remark this answers is not in the document]
Comment (thomasmck) [reply cbt_vm_metadata_import_export.py]: I think the way we have currently worded in the documentation assumes that we are just overwriting the disks of the VM we were initially tracking. Don't believe we make any mention to VMs really. Will clarify the options more in the docs.

File: cbt_import_whole_vdi.py [hunk 2]
```diff
@@ -34,15 +48,25 @@ def main():
     parser.add_argument('-p', '--password', dest='password')
     parser.add_argument('-v', '--vdi-uuid', dest='vdi_uuid')
     parser.add_argument('-f', '--filename', dest='path')
+    parser.add_argument('--new', dest='new_vdi', action='store_const',
```
Comment (robertbreker) [inline cbt_import_whole_vdi.py]: Would this be more descriptive if this parameter was called --as-new-vdi?
Or can you please specify the help-parameter?

Comment (thomasmck) [reply cbt_import_whole_vdi.py]: ok to be safe I've done both

**Pull request merged**